tests/end_to_end/end_to_end.cpp

```diff
@@ -673,3 +680,106 @@ TEST_CASE("pinned_map_enum", "[end_to_end]")
 
     ebpf_test_pinned_map_enum();
 }
+
+TEST_CASE("implicit_explicit_detach", "[end_to_end]")
+{
+    _test_helper_end_to_end test_helper;
+
+    ebpf_handle_t program_handle;
+    ebpf_handle_t map_handle;
+    uint32_t count_of_map_handle = 1;
+    uint32_t result = 0;
+    const char* error_message = NULL;
+    ebpf_handle_t link_handle;
+
+    single_instance_hook_t hook;
+    program_information_provider_t xdp_program_information(EBPF_PROGRAM_TYPE_XDP);
+
+    REQUIRE(
+        (result = ebpf_api_load_program(
+             SAMPLE_PATH "droppacket.o",
+             "xdp",
+             EBPF_EXECUTION_INTERPRET,
+             &program_handle,
+             &count_of_map_handle,
+             &map_handle,
+             &error_message),
+         error_message ? printf("ebpf_api_load_program failed with %s\n", error_message) : 0,
+         ebpf_free_string(error_message),
+         error_message = nullptr,
+         result == EBPF_SUCCESS));
```

Comment:
comma operator is hard to read here. split into separate source lines and only
REQUIRE(result == EBPF_SUCCESS)